### DISCUSSION TITLE
How to disabled the logs of requests at production env?

### DISCUSSION THREAD

**Discussion by BugMaker-Huang** [remix-run/remix#8112] [Q&A]
![image](https://github.com/remix-run/remix/assets/120626393/7b8566d1-fa02-4983-9abf-fa98d48f9eb5)

I have added a monitor SAAS to my application, and it charges based on the number of logs.
In my case, it is unnecessary to output a log every time a request is made.

**Comment by gustavopch** [comment]
Is this question really about Remix or about the monitoring tool?

**Reply by ZIA-Hans** [reply]
The monitoring tool will catch the server log, Remix will print some log in terminal when route change or fetch data.

**Comment by sergiodxa** [comment]
You need to change your HTTP server from Remix App Server to Express and then customize the logs there.

**Comment by isc30** [comment]
does any1 know how to fix achieve this?

**Reply by kiliman** [reply]
If you're using Remix App Server, it's a pre-configured Express app that uses `morgan` for logging.

If you don't want to switch to the Express template like Sergio suggested, then you can use my plugin to let you configure the Express app directly.

Here's a stackblitz
https://stackblitz.com/edit/remix-run-remix-t24cs6?file=app%2Fentry.server.tsx

Basically just add this to your _app/entry.server.tsx_ file:
```ts
export const app = createExpressApp({
  configure: (app) => app, // just return the plain express app so no middleware added
});
```

Add the `expressDevServer()` plugin in your Vite config.

https://github.com/kiliman/remix-express-vite-plugin
